.github/workflows/ci.yml

```diff
@@ -39,16 +39,45 @@ jobs:
             reqs/requirements-dev-test.txt
       - name: Install dependencies
         run: pip install -r reqs/requirements-dev-test.txt
+      - name: Get docs files that have changed
+        id: changed-files
+        uses: dorny/paths-filter@v2
+        with:
+          filters: |
+            docs:
+              - "docs/**"
+              - tests/integration/docusaurus/**
+              - tests/integration/fixtures/**
+              - tests/test_sets/**
       - name: check_repo_root_size
+        if: steps.changed-files.outputs.docs == 'true'
         run: sh ./ci/checks/check_repo_root_size.sh
       - name: Docstring linter
+        if: steps.changed-files.outputs.docs == 'true'
         run: invoke docstrings
       - name: line_number_snippet_checker
+        if: steps.changed-files.outputs.docs == 'true'
         run: python ci/checks/check_no_line_number_snippets.py
       - name: name_tag_snippet_checker
+        if: steps.changed-files.outputs.docs == 'true'
         run: python ci/checks/check_only_name_tag_snippets.py
+      - name: integration_test_gets_run_checker
+        if: steps.changed-files.outputs.docs == 'true'
+        run: python ci/checks/check_integration_test_gets_run.py
+      - name: name_tag_snippet_referenced_checker
+        if: steps.changed-files.outputs.docs == 'true'
+        run: python ci/checks/check_name_tag_snippets_referenced.py
       - name: public_api_report
+        if: steps.changed-files.outputs.docs == 'true'
         run: invoke public-api
+      - name: link_checker
+        if: steps.changed-files.outputs.docs == 'true'
+        run: python docs/checks/docs_link_checker.py -p docs/docusaurus/docs -r docs/docusaurus/docs -s docs --skip-external
+      - name: docs_build
+        env:
+          NODE_OPTIONS: --max_old_space_size=20480
```

Comment:
This is way bigger than the previous `max_old_space_size` which was `4096`. The value here is in megabytes. How much memory do we have on our github action containers?
Also, have you run this in github actions? Since you aren't changing these files here, this should be skipped on this PR?

Reply:
Oh oops I think that was a total miss on my part. 

I was able to confirm that the stage was being skipped. A few commits back it ran all the way through as well. 
![Screenshot 2023-08-08 at 2 27 33 PM](https://github.com/great-expectations/great_expectations/assets/25670697/dd521cc9-e65e-460e-87f6-6302722e1524)

Reply:
file:///var/folders/xc/q2lzqtf11m1ftwmz88783yq40000gn/T/TemporaryItems/NSIRD_screencaptureui_6fVPdg/Screenshot%202023-08-08%20at%203.01.49%20PM.png

Reply:
Confirming the check runs with the expected changes in `tests/integration/**`
![Screenshot 2023-08-08 at 3 02 10 PM](https://github.com/great-expectations/great_expectations/assets/25670697/6f07460f-e5f5-49bb-8f76-972224c36b58)